src/mlio/default_instance_reader.cxx

```diff
@@ -158,20 +159,59 @@ default_instance_reader::handle_nested_errors()
 std::optional<memory_slice>
 default_instance_reader::read_record_payload()
 {
+    if (has_corrupt_split_record_) {
+        throw corrupt_record_error(corrupt_split_error_msg_);
```

Comment:
So I'm a little confused. Isn't this caugth by the handle nested errors method? Because the error message does display which record is corrupt already.

Reply:
You are actually right. My bad. As you said this is caugth by `handle_nested_errors()`. In this case I have one ask. Please move the error message from the header file into an anonymous namespace in the implementation file and try to make the error message a little bit more generic. "cflag" is specific to RecordIO. We don't know whether another record format that we will support in the future will have the same way of splitting records.

```
namespace detail {
namespace {

constexpr const char *corrupt_split_error_msg = "...";

}  // namespace

...
```

Reply:
Fixed